### ISSUE TITLE
Remove alternate FMS & CRTM versions from unified-dev

### ISSUE THREAD

**Issue by AlexanderRichert-NOAA** [JCSDA/spack-stack#548]
Editing this issue to cover both FMS and CRTM.

FMS and CRTM currently have alternate versions in unified-dev, which, among other things, can trigger concretization bugs. It would be lovely to get to a point where we can see unify: true.

Relatedly:
The following code appears in fms/package.py:
```
    with when("@release-jcsda"):
        git = "https://github.com/JCSDA/fms.git"
```
The `with when()` construct does not function as a conditional statement, so this should be corrected as soon as possible.

**Comment by climbfuji** [comment]
Regarding 1 - this can be removed.

Regarding 2 - no idea. But with a bit of luck we'll be able to remove the release-jcsda version altogether in the next few weeks to months.

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA I don't think we'll be able to avoid multiple CRTM versions for a very long time (until everyone uses CRTM v3 - GFS v18 for the UFS?). For FMS, I am hopeful that once https://github.com/NOAA-GFDL/FMS/pull/1225 is merged, we have the opportunity to remove the different FMS versions and work with the latest FMS release in all our environments.

**Comment by climbfuji** [comment]
The FMS part of this issue is blocked by https://github.com/NOAA-GFDL/FMS/pull/1225

**Comment by climbfuji** [comment]
Kicking the can down the road to spack-stack-1.7.0 ...

**Comment by climbfuji** [comment]
@BenjaminTJohnson We added you here to check what the timeline/prospect of moving to crtm v3 everywhere is. If that's not feasible within the next year or so I suggest with close this issue **after** the FMS unification is done (hopefully soon!)

**Comment by BenjaminTJohnson** [comment]
@climbfuji do you mean to ask when we're deprecating v2.x versions and only supporting v3.x?

**Comment by climbfuji** [comment]
For spack-stack, we need to know when we can stop installing crtm@v2 with each release - probably not for a while because some operational implementations will use crtm@v2?

**Comment by BenjaminTJohnson** [comment]
@climbfuji yes, my best guess would be 2-3 years before we stop supporting v2.4.0 operationally.

**Comment by climbfuji** [comment]
We can now remove fms@release-jcsda from the templates and the JEDI packages, and have our package match spack develop. @srherbener Can you take care of this, please?

**Comment by climbfuji** [comment]
The fms divergence is now history ... all that remains is crtm.

**Comment by climbfuji** [comment]
For spack-stack-1.9.0, we still need different versions of CRTM. I will move this to the following release of spack-stack, with some luck everbody will be on the same version of CRTM 3.x.y by then.

**Comment by climbfuji** [comment]
Can't remove multiple CRTM and FMS versions, but we got the concretization bugs under control
